tsl/src/bgw_policy/job.c

```diff
@@ -89,14 +89,65 @@ get_chunk_id_to_reorder(int32 job_id, Hypertable *ht)
 															 -1);
 }
 
+static int64
+get_compression_window_end_value(Dimension *dim, Jsonb *config)
+{
+	Oid partitioning_type = ts_dimension_get_partition_type(dim);
+
+	if (IS_INTEGER_TYPE(partitioning_type))
+	{
+		int64 lag = policy_compression_get_older_than_int(config);
+		Oid now_func = ts_get_integer_now_func(dim);
+		ts_interval_now_func_validate(now_func, partitioning_type);
```

Comment:
To distinguish from variable declaration block:
```suggestion

		ts_interval_now_func_validate(now_func, partitioning_type);
```